airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -535,6 +538,20 @@ class Tickets(SourceZendeskIncrementalExportStream):
     response_list_name: str = "tickets"
     transformer: TypeTransformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
 
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[StreamData]:
+        try:
+            yield from super(Tickets, self).read_records(sync_mode, cursor_field, stream_slice, stream_state)
+        except HTTPError as e:
+            if e.response.status_code == 400 and e.response.json().get('error', '') == 'StartTimeTooRecent':
+                return []
```

Comment:
Does this error mean that we can't get any records at all, or the record that returned that error? I think `return []` will basically boot us out of the whole stream

Reply:
This is just a fix for cases when state is abnormal. All streams except this one return nothing if a date is much higher then current (2050-01-01). Tickets API returns StartTimeTooRecent error in this case which leads to raise of HTTPError error.